tests/models/test_taskinstance.py

```diff
@@ -1294,48 +1300,56 @@ def test_depends_on_past(self, dag_maker):
             pytest.param(
                 "all_done_setup_success",
                 1,
-                _UpstreamTIStates(3, 0, 1, 0, 0, 4, 1, 0),
+                [SUCCESS, SUCCESS, SUCCESS, FAILED, None, SUCCESS],
                 True,
                 (True, None),  # is_teardown=True, expect_state=None
                 False,
                 id="is teardown not all done one failed",
             ),
             pytest.param(
-                "all_done_setup_success",
+                "all_done",
                 1,
-                _UpstreamTIStates(3, 0, 1, 0, 0, 4, 1, 0),
+                [SUCCESS, SUCCESS, SUCCESS, FAILED, None, SUCCESS],
                 True,
-                (False, "upstream_failed"),  # is_teardown=False, expect_state="upstream_failed"
+                (False, None),  # is_teardown=False, expect_state=None (as one task pending)
                 False,
                 id="not teardown not all done one failed",
             ),
             pytest.param(
                 "all_done_setup_success",
                 1,
-                _UpstreamTIStates(3, 1, 0, 0, 0, 4, 1, 0),
+                [SUCCESS, SUCCESS, SUCCESS, SKIPPED, None, SUCCESS],
                 True,
                 (True, None),  # is_teardown=True, expect_state=None
                 False,
-                id="not all done one skipped",
+                id="is teardown not all done one skipped",
             ),
             pytest.param(
-                "all_done_setup_success",
+                "all_done",
                 1,
-                _UpstreamTIStates(3, 1, 0, 0, 0, 4, 1, 0),
+                [SUCCESS, SUCCESS, SUCCESS, SKIPPED, None, SUCCESS],
                 True,
-                (False, "skipped"),  # is_teardown=False, expect_state="skipped'
+                (False, None),  # is_teardown=False, expect_state=None
                 False,
-                id="not all done one skipped",
+                id="not teardown not all done one skipped",
+            ),
+            pytest.param(
+                "all_done",
+                1,
+                [SUCCESS, SUCCESS, SUCCESS, SUCCESS, SKIPPED, SUCCESS],
+                True,
+                (False, "skipped"),  # is_teardown=False, expect_state="skipped'
+                True,
+                id="not teardown all done one skipped",
```

Comment:
In this test I would have expected a skipped result but it is returning with `None`after evaluation. I assume some mocks are wrong here :-(